### DISCUSSION TITLE
Evaluate current behavior implementation

### DISCUSSION THREAD

**Discussion by lizzcondrey** [DevOps-MBSE/AaC#700] [Q&A]
As we work on building up our support for Use Cases and Behaviors of systems and their components, we want to take a look at how we currently represent and implement behavior within AaC. 

Here is the current [behavior schema](https://github.com/DevOps-MBSE/AaC/blob/a4a9b9734983fdfb5f5d3d855dc50b0f26e4ff42/python/src/aac/spec/spec.yaml#L313C1-L357C15). And here are some questions that should be answered while evaluating the current implementation.

- Does the behavior schema provide all of the necessary components for building up and representing the behavior of a system or system component? 
- Do any components need to go away? 
- Do any components need to be reworked? 
- Is the intent, usage, and role of each component understood and properly documented for our end users? 
- Is the relationship between the behavior components and Use Cases and/or feature files properly represented?
- Is that relationship properly documented?
- Is that relationship properly implemented?

**Comment by jondavid-black** [comment]
When contemplating Use Cases, I suggest we look at and build upon the work of others.  In general, a use case is a well defined thing in the world of MBSE.  Here's an overview of [use case in SysML](https://sparxsystems.com/enterprise_architect_user_guide/16.0/modeling_languages/sysml_use_cases.html#:~:text=The%20SysML%20notation%20for%20a,and%20the%20subject%20(system).).  As you can see it is a pretty simple definition.  We've focused on the Use Case scenario...mostly because that's the portion I find valuable so that's what I put in when I originally built out the aac modeling structure.

It would be helpful to identify which of the questions above are most important to users and capture some pseudo-aac representations of improvements users would find valuable.

**Comment by lizzcondrey** [comment]
This is currently being further evaluated and experimentally implemented within the AaC 4.0 work going on. So more thorough review and consideration will be able to be conducted once that experiment implementation is in place to determine any remaining holes.
